alembic_migration/versions/8c230a4a0284_add_column_feed_filter_langs_to_users.py

```diff
@@ -0,0 +1,36 @@
+"""Add column feed_filter_langs to users
+
+Revision ID: 8c230a4a0284
+Revises: d4d360ef69bc
+Create Date: 2017-09-13 15:28:23.441323
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from alembic_migration.extensions import drop_enum
+from c2corg_api.models.utils import ArrayOfEnum
+
+
+# revision identifiers, used by Alembic.
+revision = '8c230a4a0284'
+down_revision = 'd4d360ef69bc'
+branch_labels = None
+depends_on = None
+
+def upgrade():
+    langs = ('fr', 'it', 'de', 'en', 'es', 'ca', 'eu')
+    lang_enum = sa.Enum(*langs, name='lang', schema='guidebook')
+    lang_enum.create(op.get_bind(), checkfirst=False)
+    op.add_column(
+        'user',
+        sa.Column(
+            'feed_filter_langs',
+            ArrayOfEnum(lang_enum),
```

Comment:
Elsewhere `lang` is a foreign key to [guidebook.langs](https://github.com/c2corg/v6_api/blob/db2dc7412001d5d60e62f97272177db6534e0138/alembic_migration/versions/38df9393c9a9_init.py#L629):
https://github.com/c2corg/v6_api/blob/db2dc7412001d5d60e62f97272177db6534e0138/alembic_migration/versions/38df9393c9a9_init.py#L730-L732

Reply:
@tsauerwein Thanks! I have made a change https://github.com/c2corg/v6_api/pull/690/commits/f161373298fed122e534c296b14277fe3af145d8

Reply:
But I am not sure it is really possible to use such a foreign key for an ARRAY typed column.
With the current definition
```
feed_filter_langs = Column(
￼    ARRAY(String(2), ForeignKey(schema + '.langs.lang')),
    nullable=False, server_default='{}')
```
the foreign key is ignored. Most likely because it is not a supported argument for ``ARRAY()``:
http://docs.sqlalchemy.org/en/latest/core/type_basics.html#sqlalchemy.types.ARRAY

If I move the foreign key definition outside of ``ARRAY()``, I get errors such as 
> sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) foreign key constraint "feed_document_changes_langs_fkey" cannot be implemented
DETAIL:  Key columns "langs" and "lang" are of incompatible types: character varying[] and character varying.
 [SQL: 'ALTER TABLE guidebook.feed_document_changes ADD FOREIGN KEY(langs) REFERENCES guidebook.langs (lang)']

Reply:
Oh yes, sorry! I didn't notice it's an array.

The cleanest would probably be to use an enum everywhere, but that doesn't have to be done in this PR.